library/Fission/CLI/Auth.hs

```diff
@@ -57,11 +57,11 @@ removeConfigFile = do
 
 withAuth :: MonadRIO   cfg m
          => HasLogFunc cfg
-         => (UserConfig -> m (Either ClientError a))
+         => (BasicAuthData -> m (Either ClientError a))
          -> m (Either SomeException a)
 withAuth action = get >>= \case
   Right auth ->
-    action auth >>= pure . \case
+    action (toBasicAuth auth) >>= pure . \case
```

Comment:
This was a WIP branch, and stuff was being messed around intentionally. Totally get that 👍 Changing this back made life much easier.

This function no longer applied only to authentication. Here's the type signature on `master`:

```haskell
withAuth :: MonadRIO   cfg m
         => HasLogFunc cfg
         => (BasicAuthData -> m (Either ClientError a))
         -> m (Either SomeException a)
```

Changing `BasicAuthData` to `UserConfig` made life much harder, and a lot of functions read in a weird way. For example, `Auth.get` now tells you about `peers`.

We have a few options here. One is to move `Auth.get` to `UserConfig.get`. Another (preferred) way is to parse `BasicAuthData` and `NonEmpty Peers` from the same file, but do it in two passes to get two data types. We can either run the two effects, or read the file in and parse the pure data.